### ISSUE TITLE
Unimplemented `fadvise64()` fails Boost FS stack

### ISSUE THREAD

**Issue by dimakuv** [gramineproject/gramine#511]
## Description of the problem

We hit a problem with a server app that uses C++ Boost library.

Boost defines by default `BOOST_BEAST_USE_POSIX_FADVISE=1`, and there is a snippet in Boost that does something like
```c
#ifdef BOOST_BEAST_USE_POSIX_FADVISE
if (posix_fadvise( ... )) {
   ... return error ...
}
#endif
```

This results in the web server returning a 501 HTTP error because it tries to open a file-to-transmit, then fails on `posix_fadvise()` on this file and thus fails to create a proper HTTP response.

Basically, we need to implement `fadvise64()` syscall. Maybe we can just stub it for now, as in our Gramine threat model this syscall is not required to do anything meaningful anyway. @pwmarcz ?

**Comment by kailun-qin** [comment]
Agree, IMO a stub should be fine for now. BTW, I don't see any of the other SGX LibOSes (Occlum, mystikos, sgx-lkl) have actual support for `fadvise64()` neither.

**Comment by boryspoplawski** [comment]
There are multiple options to `fadvise`, but we can definitely implement some of them (or even all, someone would need to analyze) as no-op, as this would be not distinguishable for the user app.

**Comment by mkow** [comment]
#699 was just merged, so I assume this can be closed now.
